### PR TITLE
feat: adding send push notification with pubsub

### PR DESCRIPTION
**Pull request by javip97**

### CODE REVIEW

File: internal/core/domain/claim.go [hunk 1]
```diff
@@ -36,10 +36,10 @@ type Claim struct {
 	CoreClaim        CoreClaim       `json:"core_claim"`
 	MTPProof         pgtype.JSONB    `json:"mtp_proof"`
 	SignatureProof   pgtype.JSONB    `json:"signature_proof"`
-	IdentityState    *string         `json:"-"`
+	IdentityState    *string         `json:"identity_state"`
```
Comment (x1m3) [inline internal/core/domain/claim.go:39]: Probably we do not need to export it as json now, as we are only adding the ID of the claim and not all the data.

File: internal/core/domain/claim.go [hunk 2]
```diff
@@ -36,10 +36,10 @@ type Claim struct {
...
 	Status           *IdentityStatus `json:"status"`
 	CredentialStatus pgtype.JSONB    `json:"credential_status"`
-	HIndex           string          `json:"-"`
+	HIndex           string          `json:"h-index"`
```
Comment (x1m3) [inline internal/core/domain/claim.go:42]: Same as identityState. Probably not needed.

File: pkg/pubsub/pubsub.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	Publish(ctx context.Context, topic string, payload Event) error
+}
+
+// EventHandler is the type that functions that handle an event must comply.
+type EventHandler func(context.Context, Event) error
+
+// Subscriber subscribes to the pubsub topics
+type Subscriber interface {
+	Subscribe(ctx context.Context, topic string, callback EventHandler)
+	Unsubscribe(ctx context.Context, topic string) error
```
Comment (x1m3) [inline pkg/pubsub/pubsub.go:43]: I think we can remove Unsuscribe for now. Not used and not clear how to use it.

File: pkg/pubsub/redis.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+	"github.com/go-redis/redis/v8"
+
+	"github.com/polygonid/sh-id-platform/internal/log"
+)
+
+// RedisClient struct
+type RedisClient struct {
+	conn        *redis.Client
+	subscribers map[string]*redis.PubSub
```
Comment (x1m3) [inline pkg/pubsub/redis.go:16]: Probably not needed if we remove the Unsubscribe method

File: internal/core/services/notification.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+		return errors.New("sendCredentialNotification unexpected data type")
+	}
+
+	return n.sendCreateCredentialNotification(ctx, cEvent.ID)
+}
+
+func (n *notification) sendCreateCredentialNotification(ctx context.Context, credID string) error {
+	credUUID, err := uuid.Parse(credID)
+	if err != nil {
+		return err
```
Comment (x1m3) [inline internal/core/services/notification.go:48]: Log the credentiaID, etc... Any info to understand what is happening.

File: internal/core/domain/claim.go (hunk 1 above)
Comment (javip97) [reply internal/core/domain/claim.go:39]: agree, I added them as I was sending the credential in the pubsub before

Comment (javip97) [reply internal/core/domain/claim.go:39]: agree, I added them as I was sending the credential in the pubsub before

File: internal/core/services/notification.go (hunk 5 above)
Comment (javip97) [reply internal/core/services/notification.go:46]: sure, thanks!

File: pkg/pubsub/redis.go
Comment (x1m3) [inline pkg/pubsub/redis.go:1]: We should write a small test here doing a publish and a subcribe to be sure that we receive the same message we send.

**Review by x1m3 — CHANGES_REQUESTED** [review]

File: pkg/pubsub/redis.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+// Publish publishes a new topic payload
+func (rdb *RedisClient) Publish(ctx context.Context, topic string, payload Event) error {
+	return rdb.conn.Publish(ctx, topic, payload).Err()
+}
+
+// Subscribe adds a topic to the
+func (rdb *RedisClient) Subscribe(ctx context.Context, topic string, callback EventHandler) {
+	pubsub := rdb.conn.Subscribe(ctx, topic)
+
+	go func() {
```
Comment (x1m3) [inline pkg/pubsub/redis.go:31]: Listen for context cancellation, run pubsub.Close and quit the loop

**Review by x1m3 — CHANGES_REQUESTED** [review]

File: pkg/pubsub/redis.go
Comment (x1m3) [reply pkg/pubsub/redis.go:1]: We agreed on doing it in another feature.

**Review by x1m3 — APPROVED** [review]

**Review by x1m3 — APPROVED** [review]

**Review by martinsaporiti — APPROVED** [review]
LGTM

**Pull request merged**